core/domain/fs_domain_test.py

```diff
@@ -105,16 +107,108 @@ def test_independence_of_file_systems(self):
         with self.assertRaisesRegexp(IOError, 'No such file or directory'):
             fs.get('non_existent_file.png')
 
+    def test_commit(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        with self.assertRaises(NotImplementedError):
+            fs.commit(self.user_id, 'abc.png', 'file_contents')
+
+    def test_isfile(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        self.assertTrue(fs.isfile('img.png'))
+        self.assertFalse(fs.isfile('fake.png'))
+
+    def test_delete(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        with self.assertRaises(NotImplementedError):
+            fs.delete(self.user_id, 'img.png')
+
+    def test_listdir(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        with self.assertRaises(NotImplementedError):
+            fs.listdir('')
+
+
+class GcsFileSystemUnitTests(test_utils.GenericTestBase):
+    """ Tests for the GCS file system."""
+
+    def setUp(self):
+        super(GcsFileSystemUnitTests, self).setUp()
+        self.user_id = 'abc@example.com'
+
+    def test_get_and_save(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            fs = fs_domain.AbstractFileSystem(
+                fs_domain.GcsFileSystem('exploration/eid'))
+            fs.commit(self.user_id, 'abc.png', 'file_contents')
+            self.assertEqual(fs.get('abc.png'), 'file_contents')
+
+    def test_delete(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            fs = fs_domain.AbstractFileSystem(
+                fs_domain.GcsFileSystem('exploration/eid'))
+            self.assertFalse(fs.isfile('abc.png'))
+            fs.commit(self.user_id, 'abc.png', 'file_contents')
+            self.assertTrue(fs.isfile('abc.png'))
+
+            fs.delete(self.user_id, 'abc.png')
+            self.assertFalse(fs.isfile('abc.png'))
+            with self.assertRaisesRegexp(
+                IOError, r'File abc\.png .* not found'):
+                fs.get('abc.png')
+
+            with self.assertRaisesRegexp(
+                IOError, 'Image does not exist: fake_file.png'):
+                fs.delete(self.user_id, 'fake_file.png')
+
+    def test_listdir(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            fs = fs_domain.AbstractFileSystem(
+                fs_domain.GcsFileSystem('exploration/eid'))
+            fs.commit(self.user_id, 'abc.png', 'file_contents')
+            fs.commit(self.user_id, 'abcd.png', 'file_contents_2')
+            fs.commit(self.user_id, 'abc/abcd.png', 'file_contents_3')
+            fs.commit(self.user_id, 'bcd/bcde.png', 'file_contents_4')
+
+            gcs_path = '/testbed-test-resources/exploration/eid/assets/'
```

Comment:
Is this constant defined somewhere else? If so can we use that. If not, can you explain how you found it out?

Reply:
Is it possible to directly use the constant over here? Don't want to repeat constants. I think that constant should be available here (As this is a subclass)? Could you try that out?

Reply:
Done.